sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/client/ViewCallValidator.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.javasdk.impl.client
+
+import java.lang.reflect.Method
+
+import kalix.javasdk.action.Action
+import kalix.javasdk.eventsourcedentity.EventSourcedEntity
+import kalix.javasdk.valueentity.ValueEntity
+import kalix.javasdk.workflow.Workflow
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestParam
+
+object ViewCallValidator {
+
+  def validate(method: Method): Unit = {
+    val declaringClass = method.getDeclaringClass
+    if (classOf[Action].isAssignableFrom(declaringClass)
+      || classOf[ValueEntity[_]].isAssignableFrom(declaringClass)
+      || classOf[EventSourcedEntity[_, _]].isAssignableFrom(declaringClass)
+      || classOf[Workflow[_]].isAssignableFrom(declaringClass)) {
+      throw new IllegalStateException(
+        "Use dedicated builder for calling " + declaringClass.getSuperclass.getSimpleName
+        + " component method " + declaringClass.getSimpleName + "::" + method.getName + ". This builder is meant for View component calls.")
```

Comment:
I just added a comment about this. 

Instead, we could check that the method is annotated with `@Query`. That will also cover multi-table views.